runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -313,6 +313,15 @@ public synchronized ILogData previous() {
 
         // If still null, we're done.
         if (prevAddress == null) {
+            if (context.checkpointSuccessId == null) {
```

Comment:
Please revert the outer if and avoid nested one.

Reply:
Done.